klippy/extras/bed_mesh.py

```diff
@@ -584,6 +590,98 @@ def update_config(self, gcmd):
 
         params = gcmd.get_command_parameters()
         need_cfg_update = False
+
+        # If adaptive meshing is enabled and there are defined objects,
+        # generate adaptive mesh
+        if do_adaptive and exclude_object is not None and \
+                len(exclude_object.objects) != 0:
```

Comment:
Why isn't it valid? `ExcludeObject._reset_state()` is being called from the constructor so the member attribute should always be defined. I can switch the code as your suggestion gives me the same result but I just want to understand your comment.

Reply:
It's against our coding conventions as mentioned at: https://www.klipper3d.org/Code_Overview.html#adding-a-host-module

Accessing the internal member variables of other objects leads to a maintenance burden.  Other developers working on ExcludeObject wont know that some other module has accessed its interanl state and those developers may make a local change that could subtly break the bed_mesh module.  So, the convention is to avoid doing that.

-Kevin